channels/rdpdr/server/rdpdr_main.c

```diff
@@ -252,48 +318,76 @@ static int rdpdr_server_read_smartcard_capability_set(RdpdrServerContext* contex
 static int rdpdr_server_write_smartcard_capability_set(RdpdrServerContext* context, wStream* s)
 {
 	RDPDR_CAPABILITY_HEADER header;
+
 	header.CapabilityType = CAP_SMARTCARD_TYPE;
 	header.CapabilityLength = RDPDR_CAPABILITY_HEADER_LENGTH;
 	header.Version = SMARTCARD_CAPABILITY_VERSION_01;
-	if (!Stream_EnsureRemainingCapacity(s, header.CapabilityLength))
-		return -1;
+
+	Stream_EnsureRemainingCapacity(s, header.CapabilityLength);
+
 	rdpdr_server_write_capability_set_header(s, &header);
+
 	return 0;
 }
 
 static int rdpdr_server_send_core_capability_request(RdpdrServerContext* context)
 {
 	wStream* s;
-	BOOL status = FALSE;
+	BOOL status;
 	RDPDR_HEADER header;
 	UINT16 numCapabilities;
 	ULONG written;
-	WLog_DBG(TAG, "RdpdrServerSendCoreCapabilityRequest");
+
+	WLog_DBG(TAG, "RdpdrServerSendCoreCapabilityRequest"); fflush(stdout);
```

Comment:
I think we shouldn't add fflush to the code directly since it is possible that WLog_DBG doesn't use stdout at all (for example when a file is used for logging. 
A possible solution would be to either use stderr for logging in this case or add the fflush in Console appender.